### DISCUSSION TITLE
How to run a thread in socket.on()

### DISCUSSION THREAD

**Discussion by SwitchGM** [miguelgrinberg/Flask-SocketIO#2071] [Q&A]
Hello all,

Sorry for what may be a very simple question. 

I have recently started implementing threads into my flask socketio application, and am running into a problem when those threads are running.

If I start a thread that runs forever, it appears to take over the main application thread, and subsequent lines do not run.

```python
from flask import Flask
from flask_socketio import SocketIO
from threading import Thread


app = Flask(__name__)
socketio = SocketIO(app)

@socketio.on("test")
def handle_test():
    def run():
        while True:
            ...
    
    print("i see this line")
    Thread(target=run).start()
    print("i don't see this line")

socketio.run(app, port=3000)
```

When running handle_test without flask-socketio, I see both lines and the thread runs just fine. 

But when I run the thread in the above setup, I only see the first line and my program jumps right into the thread instead of running the second line.

I assume that I'm using threads wrong in this situation and/or there's some config option that I need to set.

I've come across a few of the repo's issues that may be related to my issue, but I am struggling to understand their relevance to my situation!

https://github.com/miguelgrinberg/Flask-SocketIO/issues/1990
https://github.com/miguelgrinberg/Flask-SocketIO/issues/1801
https://github.com/miguelgrinberg/Flask-SocketIO/issues/410

Any help with this is greatly appreciate!

**Comment by miguelgrinberg** [comment]
Are you using eventlet or gevent in this application? These frameworks are not compatible with standard threads, you have to monkey patch the standard library if you want to use them, or else use the native concurrency they provide.

**Reply by SwitchGM** [reply]
Thank you for the response :) 

I'm not aware of or familiar with either. How would I go about finding out which I am using? (I don't believe that I have imported either into my application)

**Reply by SwitchGM** [reply]
After a bit of searching through the mud, it looks like I'm using gevent :)

**Reply by miguelgrinberg** [reply]
Okay. Try uninstalling it.
